src/pjs/pjs.proto

```diff
@@ -0,0 +1,184 @@
+syntax = "proto3";
+
+package pjs;
+option go_package = "github.com/pachyderm/pachyderm/v2/src/pjs";
+
+import "google/protobuf/any.proto";
+
+message Job {
+  int64 id = 1;
+}
+
+enum JobState {
+  // UNSPECIFIED means the job state is unspecified.
+  JOB_STATE_UNSPECIFIED = 0;
+  // QUEUED means the job is currently in a queue.
+  // A QUEUED job will not have any descendants.
+  QUEUED = 1;
+  // PROCESSING means the job is currently being processed by a worker.
+  PROCESSING = 2;
+  // DONE means the job, and all of its descendants, are done.
+  DONE = 3;
+}
+
+enum JobErrorCode {
+  // UNSPECIFIED means the job error code is unspecified.
+  JOB_ERROR_CODE_UNSPECIFIED = 0;
+  // FAILED means that the worker processing the job indicated that it failed.
+  FAILED = 1;	
+  // DISCONNECTED means the worker processing the job disconnected.
+  DISCONNECTED = 2;
+  // CANCELED means the job was canceled.
+  CANCELED = 3;
+}
+
+message JobInfo {
+  Job job = 1; 
+  Job parent_job = 2;
```

Comment:
I assume that there can be parentless jobs, so this is probably a case where a nil Job is the right identifier.  But we should be sure to reject a zero-valued job in that case.